cognite/neat/core/_instances/loaders/_rdf_to_instance_space.py

```diff
@@ -0,0 +1,234 @@
+            if cdf_space is None:
+                to_create.append(local_space)
+            elif cdf_space != local_space.as_write():
+                to_update.append(local_space)
+            else:
+                unchanged.append(local_space)
+        loader = client.loaders.spaces
+        results: UploadResult[str] = UploadResult("instance spaces")
+        results.unchanged.update(unchanged.as_ids())
+        if dry_run:
+            results.created.update(to_create.as_ids())
+            results.changed.update(to_update.as_ids())
+            yield results
+        if to_create:
+            try:
+                client.loaders.spaces.create(to_create)
+            except MultiCogniteAPIError as e:
+                results.failed_created.update(to_create.as_ids())
+                for error in e.errors:
+                    results.error_messages.append(f"Failed to create {loader.resource_name}: {error!s}")
+            else:
+                results.created.update(to_create.as_ids())
+
+        if to_update:
+            try:
+                client.loaders.spaces.update(to_update)
+            except MultiCogniteAPIError as e:
+                results.failed_changed.update(to_update.as_ids())
+                for error in e.errors:
+                    results.error_messages.append(f"Failed to update {loader.resource_name}: {error!s}")
+            else:
+                results.changed.update(to_update.as_ids())
+
+        yield results
+
+    def write_to_file(self, filepath: Path) -> None:
+        """Dumps the instance spaces to file."""
+        if filepath.suffix not in [".json", ".yaml", ".yml"]:
+            raise ValueError(f"File format {filepath.suffix} is not supported")
+        dumped: dict[str, list] = {"spaces": [], "issues": []}
+        for item in self.load(stop_on_exception=False):
+            key = {
+                dm.SpaceApply: "spaces",
+                NeatIssue: "issues",
+            }.get(type(item))
+            if key is None:
+                # This should never happen, and is a bug in neat
+                raise ValueError(f"Item {item} is not supported. This is a bug in neat please report it.")
+            dumped[key].append(item.dump())
+        with filepath.open("w", encoding=self._encoding, newline=self._new_line) as f:
+            if filepath.suffix == ".json":
+                json.dump(dumped, f, indent=2)
+            else:
+                yaml.safe_dump(dumped, f, sort_keys=False)
+
+    def _load(
+        self, stop_on_exception: bool = False
+    ) -> Iterable[dm.SpaceApply | NeatIssue | type[_END_OF_CLASS] | _START_OF_CLASS]:
+        yield from self._lookup_issues
+        seen: set[str] = set()
+        for space_str in self.space_by_instance_uri.values():
+            if space_str in seen or space_str in COGNITE_SPACES:
+                continue
+            yield dm.SpaceApply(space=space_str)
+            seen.add(space_str)
```

Comment:
If you have very large number of instances maybe it is quicker to execute set operation over values

```suggestion
        for space_str in set(self.space_by_instance_uri.values()):
            if space_str in seen or space_str in COGNITE_SPACES:
                continue
            yield dm.SpaceApply(space=space_str)
            seen.add(space_str)
```